packages/nextra/src/context.ts

```diff
@@ -24,12 +23,12 @@ function filter(
 } {
   let activeLevelPages: Page[] = []
   const items: Page[] = []
-  const meta =
-    pageMap.find((item): item is MetaJsonFile => item.name === META_FILENAME)
-      ?.meta || {}
+  const meta = pageMap.find(
+    (item): item is MetaJsonFile => item.kind === 'Meta'
+  )!.data
 
   for (const item of pageMap) {
-    if (item.name === META_FILENAME) continue
+    if (item.kind === 'Meta') continue
```

Comment:
I explicitly added `kind` property, so TypeScript will be much smarter to understand of which kind of file is processed